raiden/network/blockchain_service.py

```diff
@@ -137,14 +139,27 @@ def token(self, token_address: Address) -> Token:
 
         with self._token_creation_lock:
             if token_address not in self.address_to_token:
-                self.address_to_token[token_address] = Token(
+                token = Token(
                     jsonrpc_client=self.client,
                     token_address=token_address,
                     contract_manager=self.contract_manager,
                 )
 
+                self.check_token_has_supply(token)
+                self.address_to_token[token_address] = token
+
         return self.address_to_token[token_address]
 
+    @staticmethod
+    def check_token_has_supply(token: Token):
```

Comment:
i wanted to include this function in `raiden/network/rpc/client.py` since it is conceptually similar to `check_address_has_code`. 

the issue was that we don't need a client object to execute this code; i took that as a hint that a better location should be found for this.

i also opted not to have this in the `Token` class. if we were to place it there, it should be in the constructor—we wouldn't want an object which raises an exception on this function—and implementing that means too many changes for it to be worth it atm.

so i think this location is a good middle ground